source/Calamari/Azure/AzureClient.cs

```diff
@@ -7,14 +7,20 @@ internal static class AzureClient
     {
         public static IAzure CreateAzureClient(this ServicePrincipalAccount servicePrincipal)
         {
+            AzureEnvironment GetAzureEnvironment(string environmentName)
+            {
```

Comment:
I'm not 100% sure if this is the behaviour we want here. It feels like if it isn't either one of the names expected, or "AzureCloud" _exactly_, we should throw an exception with a clear message.

A comment with a reference to the class in the core sashimi account repo that does the "fixing" of the default name to  "AzureCloud" will also help future people understand why this compensation is in here.

Reply:
What I wanted to do is lift that assumption about "AzureCloud" higher up, so that by the time we get here we receive a definitive value or null/empty. Adding another place that deals with a magic "AzureCloud" string just doesn't feel right.

Throwing an exception makes sense.

Reply:
Yeah I understand - the real solution would be mapping to our own stable model set to contain these invariants. But for now I think propagating the magic string and providing a link to its origins is acceptable. Unfortunately the origin is in Sashimi, whereas this is Calamari, and the two don't have a common set of dependencies that could house a shared definition.